src/Ganeti/JQueue.hs

```diff
@@ -562,6 +563,33 @@ isQueuedJobDead ownlivelock =
   . mfilter (/= ownlivelock)
   . qjLivelock
 
+-- | Waits for a job's process to exit
+waitUntilJobExited :: Livelock -- ^ LuxiD's own livelock
+                   -> JobId -- ^ the job to wait for
+                   -> Int -- ^ timeout in milliseconds
+                   -> ResultG (Bool, String)
+waitUntilJobExited luxiLivelock jid tmout = do
+  let sleepDelay = 100 :: Int -- ms
+      tries = tmout `div` sleepDelay
+      jName = ("Job " ++) . show. fromJobId $ jid
+  logDebug $ "Waiting for " ++ jName ++ " to exit"
+  result <- runMaybeT . msum . flip map [0..tries :: Int] $ \tryNo -> do
+    -- if we're retrying, sleep for some time
+    when (tryNo > 0) . liftIO . threadDelay $ sleepDelay * 1000
+
+    qDir <- liftIO queueDir
+    (job, _) <- lift . mkResultT $ loadJobFromDisk qDir True jid
```

Comment:
If I understand correctly, you're loading and parsing the job from disk every 100ms. My (weak) understanding is that a job's livelock cannot change once it has its own livelock, in which case it would be better to a) load the job once, and then b) pass the same job to `isQueueJobDead`, or directly to `isDead`?

Reply:
Yes, I didn't like loading the job every 100ms either. I'll refactor the loop a bit, thanks!